tasks/discover-granules/index.js

```diff
@@ -231,12 +233,16 @@ const checkGranuleHasNoDuplicate = async (granuleId, duplicateHandling) => {
  *
  * @param {string[]} granuleIds - Array of granuleIds to filter
  * @param {string} duplicateHandling - flag that defines this function's behavior (see description)
+ * @param {number} concurrency - pMap concurrency
  *
  * @returns {Array.string} returns granuleIds parameter with applicable duplciates removed
  */
-const filterDuplicates = async (granuleIds, duplicateHandling) => {
-  const keysPromises = granuleIds.map((key) =>
-    checkGranuleHasNoDuplicate(key, duplicateHandling));
+const filterDuplicates = async (granuleIds, duplicateHandling, concurrency) => {
+  const keysPromises = await pMap(
```

Comment:
Hmm - does this make sense with the line below that does a `Promise.all`? Previously we were doing a `map` on granuleIds to prepare an array of promises, then awaiting them all with `Promise.all`

But now we're explicitly awaiting the pMap operation and the mapped promises, so I think the Promise.all below is unnecessary? Am I missing something?

Reply:
I don't think you are missing something. Will push a change.

Reply:
https://github.com/nemreid/cumulus/commit/c4e5faa5cfd6182086a7e45b7f1d99f32ac1b5cf